### DISCUSSION TITLE
Files Service - msword MIME type management

### DISCUSSION THREAD

**Discussion by andredev93** [mia-platform/community#348] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

2.9.1

### Description

Hi,

we are using the **files-service** to upload files for one client, on an on-premise installation of the Console (v11.7.1).

We have to upload some old _doc_ MS Word files (97-2003 version) but the service responds with HTTP 415 with this error: _Invalid file provided, content type is not accepted by include list_. The service worked normally on _docx_ files.

The env FILE_TYPE_INCLUDE_LIST is configured to contain the _doc_ extension.

The uploaded file extension is _doc_ while the MIME reported locally by the _file_ command and a python script using magic library is _application/msword_.

But from the service logs there are reported these different MIMEs that cause the error:
- from the file extension: _application/msword_
- from the file stream: _application/x-cfb_

We also configured the ADDITIONAL_MIME_TYPES env with the `doc:application/x-cfb` value but in this way the service works for the doc files but fails on the on _docx_ files with 415 error.

Is there a way to manage both old and new msword files for the upload?

Thanks.

### Actual Outcome

The service returns error 415 on doc files while uploads correctly docx files.

### Expected Outcome

The service should upload both doc and docx files.

**Comment by pmarchini** [comment]
Hey @andredev93,

In order to create a reproducible can you share `FILE_TYPE_INCLUDE_LIST` and `ADDITIONAL_MIME_TYPES` you're using at the moment?

**Reply by andredev93** [reply]
Hi @pmarchini,

we tested with `FILE_TYPE_INCLUDE_LIST: doc` and `ADDITIONAL_MIME_TYPES` not set to have the 415 error on the _doc_ files.

**Reply by andredev93** [reply, marked as answer]
Hi @pmarchini 

actually I have to amend this remark:

> We also configured the ADDITIONAL_MIME_TYPES env with the doc:application/x-cfb value but in this way the service works for the doc files but fails on the on docx files with 415 error.

It actually works like this. Earlier we probably had a different error depending on some other factor.

During the analysis we thought if it is possible to put the MIME check under a flag to disable it when not needed, for example during massive upload of files we would like to disable the checks on file format and MIME and enabling them again when it is the final user to upload files as security measure.

Thanks.
